warehouse/oidc/forms.py

```diff
@@ -122,21 +117,17 @@ def _lookup_owner(self, owner):
 
         return response.json()
 
-    def validate_owner(self, field):
-        owner = field.data
-
-        # We pre-filter owners with a regex, to avoid loading GitHub's API
-        # with usernames/org names that will never be valid.
-        if not _VALID_GITHUB_OWNER.match(owner):
-            raise wtforms.validators.ValidationError(
-                _("Invalid GitHub user or organization name.")
-            )
+    def validate_repo_slug(self, field):
+        # Compute DB's `repository_name` `repository_owner` from the form's `repo_slug`.
+        repo_slug = field.data.split("/")
+        (owner, repository) = (repo_slug[0], repo_slug[1])
 
         owner_info = self._lookup_owner(owner)
 
         # NOTE: Use the normalized owner name as provided by GitHub.
         self.normalized_owner = owner_info["login"]
         self.owner_id = owner_info["id"]
+        self.normalized_repository = repository
```

Comment:
Nitpick: We haven't actually done any normalization here, so let's just call this `self.repository` 🙂

Reply:
(There probably _is_ some normalization we could do here, but it isn't super well defined: GitHub URLs are _somewhat_ case sensitive, but I don't think the repository name in the claim is normalized in any way.)